components/ca-treeview.js

```diff
@@ -0,0 +1,263 @@
+define(
+['/components/helpers/create-element.js', '/components/helpers/parent-by-attribute.js', '/components/helpers/cancel-event.js', 'document-register-element'],
+(createElement, parentByAttribute, cancelEvent) => {
+    /**
+     * Class TreeView for a TreeView web component
+     * @extends HTMLElement
+     */
+    class TreeView extends HTMLElement {
+        /**
+         * createdCallback is called when the component is first created.
+         * creates children, binds events and loads data if src is avaliable.
+         * @returns {undefined} nothing.
+         */
+        createdCallback() {
+            this.nav = createElement(this, 'nav', {});
+            this.ol = createElement(this.nav, 'ol');
+            this.nav.onclick = this.eventDelegate.bind(this);
+
+            if (this.dataSrc) {
+                this.loadData();
+            }
+        }
+
+        /**
+         * Called every time an attribute is changed.
+         * @param {string} attrName, the name of the changed attribute.
+         * @param {string} oldVal, previous value associated with attribute.
+         * @param {string} newVal, current value associated with attribute.
+         * @returns {undefined} if data-src or data-root change, then trigger data fetch.
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+            switch (attrName) {
+                case 'data-src': {
+                    if (newVal !== '') {
+                        this.loadData();
+                    }
+                } break;
+
+                case 'data-root': {
+                    this.root = newVal;
+                } break;
+
+                default: break;
+            }
+        }
+
+        /**
+         * Gets the root attribute
+         * @returns {string} data-root attribute, default is empty.
+         */
+        get root() {
+            return this.getAttribute('data-root') || '';
+        }
+
+        /**
+         * Gets the root attribute
+         * @param {string} value, data-root attribute, default is empty.
+         */
+        set root(value = '') {
```

Comment:
This default value is not equivalent to the previous version. It's not even just a refactoring, it's a behaviour change. It used to say:
`                this.setAttribute('data-root', value || '');`
which would assign an empty string for any falsy value. In particular, passing null will behave differently.